src/main.cpp

```diff
@@ -2906,7 +2922,7 @@ int MainFrame::txCallback(
 
     assert(framesPerBuffer < MAX_FPB);
 
-    if (rptr && !endingTx) {
+    if (rptr && !endingTx && !(g_playFileToMicIn && (g_sfPlayFile != NULL))) {
```

Comment:
Can we pls use a design that avoids extra conditionals?  I'm feel this (and the extra fifo write step) is more complicated than the original design of the mic-in system.

The original design was:

1. Read infifo to buf
1. Optionally replace buf samples with mic-in samples from file
1. (Add new resampler here)
1. Speex noise processor

Is there any reason why the original design wouldn't work?

Reply:
I think this was something I was needing to do to avoid injecting the mic audio in place of the voice keyer audio. Moving some code around in txRxProcessing() avoids the need for this, though, so I'll remove.